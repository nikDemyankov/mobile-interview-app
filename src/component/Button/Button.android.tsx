import React, { memo, useMemo } from 'react';
import { Text, ViewStyle, TouchableNativeFeedback, View } from 'react-native';

import styles from './Button.styles';

export interface Props {
  onPress: () => void;
  label: string;
  style?: ViewStyle;
}

const Button: React.FC<Props> = ({ onPress, label, style }) => {
  const buttonStyles = useMemo(() => [styles.button, style], [style]);

  return (
    <TouchableNativeFeedback onPress={onPress}>
      <View style={buttonStyles}>
        <Text style={styles.label}>{label}</Text>
      </View>
    </TouchableNativeFeedback>
  );
};

export default memo(Button);
