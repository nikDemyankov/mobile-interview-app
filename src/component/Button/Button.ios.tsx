import React, { memo, useMemo } from 'react';
import { Text, ViewStyle, TouchableOpacity } from 'react-native';

import styles from './Button.styles';

export interface Props {
  onPress: () => void;
  label: string;
  style?: ViewStyle;
}

const Button: React.FC<Props> = ({ onPress, label, style }) => {
  const buttonStyles = useMemo(() => [styles.button, style], [style]);

  return (
    <TouchableOpacity onPress={onPress} style={buttonStyles}>
      <Text style={styles.label}>{label}</Text>
    </TouchableOpacity>
  );
};

export default memo(Button);
