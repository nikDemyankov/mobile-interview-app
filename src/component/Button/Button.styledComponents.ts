import styled from 'styled-components/native';
import { color } from '../../ui/theme';

export const Label = styled.Text({
  color: color.background,
  fontWeight: 'bold',
  paddingVertical: 10,
  paddingHorizontal: 14,
  fontSize: 16,
  textAlign: 'center',
});

const buttonStyle = {
  backgroundColor: color.primary,
  borderRadius: 6,
};
export const TouchableButton = styled.TouchableOpacity(buttonStyle);
export const ButtonContainer = styled.View(buttonStyle);
