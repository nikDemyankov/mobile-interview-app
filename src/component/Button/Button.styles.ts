import { StyleSheet } from 'react-native';
import { color } from '../../ui/theme';

const styles = StyleSheet.create({
  button: {
    backgroundColor: color.primary,
    borderRadius: 6,
  },
  label: {
    color: color.background,
    fontWeight: 'bold',
    paddingVertical: 10,
    paddingHorizontal: 14,
    fontSize: 16,
    textAlign: 'center',
  },
});

export default styles;
