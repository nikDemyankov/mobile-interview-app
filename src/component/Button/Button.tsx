import React, { memo } from 'react';
import { StyleProp, ViewStyle } from 'react-native';
import { TouchableButton, Label } from './Button.styledComponents';

export interface Props {
  onPress: () => void;
  label: string;
  style?: StyleProp<ViewStyle>;
}

const Button: React.FC<Props> = ({ onPress, label, style }) => {
  return (
    <TouchableButton onPress={onPress} style={style}>
      <Label>{label}</Label>
    </TouchableButton>
  );
};

export default memo(Button);
