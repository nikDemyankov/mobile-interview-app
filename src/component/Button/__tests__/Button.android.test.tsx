import React from 'react';
import { TouchableNativeFeedback } from 'react-native';
import { act, create } from 'react-test-renderer';

import Button, { Props } from '../Button.android';

let props: Props;

beforeEach(() => {
  props = {
    onPress: jest.fn(),
    label: 'Hello',
  };
});

it('should render properly', () => {
  const component = create(<Button {...props} />);
  expect(component).toMatchSnapshot();
});

it('should render properly with style', () => {
  const component = create(<Button {...props} style={{ backgroundColor: 'red' }} />);
  expect(component).toMatchSnapshot();
});

describe('when pressed', () => {
  it('should call onPress', () => {
    const component = create(<Button {...props} />);

    expect(props.onPress).not.toHaveBeenCalled();
    act(() => {
      component.root.findByType(TouchableNativeFeedback).props.onPress();
    });
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });
});
