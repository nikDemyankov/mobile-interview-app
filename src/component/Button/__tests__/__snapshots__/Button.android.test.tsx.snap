// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#7173ce",
        "borderRadius": 6,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 16,
        "fontWeight": "bold",
        "paddingHorizontal": 14,
        "paddingVertical": 10,
        "textAlign": "center",
      }
    }
  >
    Hello
  </Text>
</View>
`;

exports[`should render properly with style 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#7173ce",
        "borderRadius": 6,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 16,
        "fontWeight": "bold",
        "paddingHorizontal": 14,
        "paddingVertical": 10,
        "textAlign": "center",
      }
    }
  >
    Hello
  </Text>
</View>
`;
